lib/github.sh

```diff
@@ -138,88 +138,37 @@ strap::github::api::token::create() {
 
   local -r machine_desc="$(strap::os::model)"
 
-  while [[ -z "$token" && ${password_attempts} < ${max_password_attempts} ]]; do
+  while [[ -z "$token" ]]; do
+  
+    if [ "$(uname -s)" == "Darwin" ]; then
```

Comment:
Should call `uname -s` and save it to a variable to ensure that we don't shell out multiple times.  That is:

```bash
local uname=''
uname="$(uname -s)"
# if/elif here
```

Reply:
is there a reason why this needs to be on two different lines? can it just be:

`local uname="$(uname -s)"`

Reply:
Declaring a variable with assignment and executing a shell command on the same line could be problematic - if either the shell command failed, or the assignment failed, you wouldn't know which of the two failed.  Having the two on separate lines ensures you can find out which line failed and why.

See https://google.github.io/styleguide/shellguide.html#s7.6-use-local-variables for more:

> Declaration and assignment must be separate statements when the assignment value is provided by a command substitution; as the local builtin does not propagate the exit code from the command substitution.

HTH!

Reply:
makes sense! i've pushed the fixes. I set the variable once at the beginning of the function. Is it ok this way, or should i set the variable each time before the two different if-else blocks?

Reply:
At the beginning of the function is usually fine. If it is used a lot, or you're not sure, it's ok to clear it before setting it later in the function